src/resources/stores.jl

```diff
@@ -21,43 +20,47 @@ See [`Container`](@ref) for a more lock-like resource.
 
 Think of `Resource` and `Container` as locks and of `Store` as channels. They block only if empty (on taking) or full (on storing).
 """
-mutable struct Store{T} <: AbstractResource
+mutable struct Store{N, T<:Number} <: AbstractResource
   env :: Environment
   capacity :: UInt
   load :: UInt
-  items :: Dict{T, UInt}
+  items :: Dict{N, UInt}
   seid :: UInt
-  put_queue :: DataStructures.PriorityQueue{Put, StorePutKey{T}}
-  get_queue :: DataStructures.PriorityQueue{Get, StoreGetKey}
-  function Store{T}(env::Environment; capacity=typemax(UInt)) where {T}
-    new(env, UInt(capacity), zero(UInt), Dict{T, UInt}(), zero(UInt), DataStructures.PriorityQueue{Put, StorePutKey{T}}(), DataStructures.PriorityQueue{Get, StoreGetKey}())
```

Comment:
The explicit `UInt(capacity)` cast is removed in the new version. That cast is necessary in order to make `capacity=10` work as a keyword argument (which used to be the case until this PR).

Reply:
SimJulia.jl had the `UInt` signature in the `capacity` kwarg. I like the idea of using `UInt(capacity)` instead of imposing this type on the kwarg, making it possible to use `capacity=10` as you suggest. However, we need to still convert to `UInt` so that a `Store` cannot have a negative capacity.